test/files/scalacheck/MutableTreeMap.scala

```diff
@@ -0,0 +1,249 @@
+import java.io._
+
+import org.scalacheck._
+import org.scalacheck.Arbitrary._
+import org.scalacheck.Prop.forAll
+
+import scala.collection.generic.CanBuildFrom
+import scala.collection.mutable
+import scala.util.Try
+
+trait MutableTreeMapGenerators {
+
+  // Note: in scalacheck 1.12.2 tree maps can be automatically generated without the need for custom
+  // machinery
+  def genTreeMap[A: Arbitrary: Ordering, B: Arbitrary]: Gen[mutable.TreeMap[A, B]] = {
+    import org.scalacheck.Gen._
+    for {
+      keys <- listOf(arbitrary[A])
+      values <- listOfN(keys.size, arbitrary[B])
+    } yield mutable.TreeMap(keys zip values: _*)
+  }
+
+  implicit def arbTreeMap[A: Arbitrary: Ordering, B: Arbitrary] = Arbitrary(genTreeMap[A, B])
+}
+
+object MutableTreeMapProperties extends Properties("mutable.TreeMap") with MutableTreeMapGenerators {
```

Comment:
I could do that, but two problems arise:

* Right now, the `RedBlackTree` object is `private[collection]`, which means that I don't have access to that method in this file. I can easily declare this file to be in package `scala.collection.mutable` though, as some of the other scalacheck tests do, e.g. `avl.scala`;
* More importantly, I can't access the `RB.Tree` inside a `mutable.TreeMap` as it is private, and I think making it `private[collection]` just for this use case maybe isn't worth it.

For the second problem, what do you propose? I can implement another set of scalacheck properties where I test inserting and removing multiple elements on a `RB.Tree`, checking if it `isValid` on each step. I can also do mixed inserts/deletes on `mutable.TreeMap` and check if its iterator returns the correct entries on each step. I like the second approach more, as I tend to prefer blackbox testing on situations like this - trust that scalacheck finds the sequence of public operations that would break the internals, which leads to externally visible bugs (e.g. missing or duplicate entries). But I also see advantages on the whitebox approach and I can surely do tests directly on `RedBlackTree` if you want.